src/cbmc_starter_kit/template-for-repository/proofs/lib/summarize.py

```diff
@@ -0,0 +1,113 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import json
+import logging
+import sys
+
+
+def _get_max_length(column):
+    return max(len(word) for word in column)
+
+
+def _get_list_of_max_length_per_column(data):
+    list_of_max_length_per_column = []
+    num_rows = len(data)
+    num_columns = len(data[0])
+    for column_idx in range(num_columns):
+        column = [data[row_idx][column_idx] for row_idx in range(num_rows)]
+        max_length = _get_max_length(column)
+        max_length_with_extra_space = max_length + 1
+        list_of_max_length_per_column.append(max_length_with_extra_space)
+    return list_of_max_length_per_column
+
+
+def _get_header(list_of_max_length_per_column, header_names):
```

Comment:
I don't understand why this is a separate function. Why not make the header names the first item in the table? They don't appear to be formatted any differently than any of the other items. I would just use `_get_entries` to iterate over all rows including the headers, and then insert a line of dashes in `data[1]`.

Reply:
Good point. Removed.